playbooks/templates/standalone_parameters.yaml.j2

```diff
@@ -81,6 +81,12 @@ parameter_defaults:
   CephPoolDefaultPgNum: 8
   CephPoolDefaultSize: 1
 {% endif %}
+{% if rhsm_enabled %}
+  ContainerImageRegistryCredentials:
+    registry.redhat.io:
+      {{ rhsm_username | mandatory }}: "{{ rhsm_password | mandatory }}"
```

Comment:
This is going to require people to put rhsm_password in a plain text file. I don't think we can do this. I know other RH services require similar, but we shouldn't add to the mess.

What other ways are there to pass these credentials? Reading from a file descriptor when the process is launched would be ideal, but even something like an environment variable would be better than persisting it in plain text.

Reply:
This is the way it's implemented in TripleO, I'm not sure how you would do otherwise unless we use something like ansible-vault or something.

Reply:
I was actually just thinking about local-overrides.yaml, but of course this is also going to write it into the template.

That's really bad design. Can we generate a pull secret that contains single-purpose credentials here? Or is plain text authentication the only supported method? Plain text passwords doesn't sound like something that would fly with many customers, so I'm convinced there's another way to do this.

Reply:
Our customers are writing Heat environments with RHSM credentials since OSP 7 I believe. I'm not sure what to say. Yes this is bad design but I don't think we can solve it here.
I might have another alternative proposal where we don't use ContainerImageRegistryCredentials to login against the container registries, but manually run "podman login" during the deployment (using Ansible in dev-install), so we only "store" credentials in dev-install overrides (which can be encrypted via git-crypt) and not in clear in the Heat environments on the host where OpenStack is deployed.

If we go that route, we'll have to test it (I think it'll work but not 100% sure).

Reply:
I updated it to have a new parameter `redhat_registry_credentials` that will be used exclusively for RH container registry credentials.